storage/database.go

```diff
@@ -165,12 +168,14 @@ func NewDatabase(
 		return nil, err
 	}
 
-	// wait till first value is received
+	// Wait till first namespaces value is received and set the value.
+	// Its important that this happens before the mediator is started to prevent
```

Comment:
Might make sense to move the mediator construction below the namespace calls, what do you think?

Reply:
I think practically it has no effect (because tick routines don't start until Open() is called), but conceptually it makes sense yeah.

Also in the terminate method we briefly discussed re-ordering things so the mediator is closed and then we close the namespace watcher instead of the opposite. Again, I don't think it has any effect but might make more conceptual sense. What do you think?

Reply:
Yeah makes sense. Could you do the reordering in Close/Terminate too.

Reply:
cool done